docs/eq_snapshot.md

```diff
@@ -19,9 +19,154 @@ Example:
         assert 2 + 2 == snapshot(4)
     ```
 
+=== "value changed"
+    <!-- inline-snapshot: outcome-failed=1 -->
+    ```python
+    def test_something():
+        assert 2 + 3 == snapshot(4)
+    ```
+
+=== "--inline-snapshot=fix"
+    <!-- inline-snapshot: fix -->
+    ```python
+    def test_something():
+        assert 2 + 3 == snapshot(5)
+    ```
+
+The value can later be changed with `--inline-snapshot=fix` if the value the snapshot is compared with has changed.
+
+## dirty-equals
```

Comment:
I think it makes sense to mention dirty-equals because it's a good use case for this feature and they make a great combination. But I would suggest first explaining the general feature so that people don't have to learn about dirty-equals. Here's an example of how it can make tests more readable:

```python
def summary(*values):
    return {"total": sum(values)}


def test_summary():
    assert summary(1, 2, 3) == snapshot({"total": 1 + 2 + 3})  # clearer than {"total": 6}
```

Here's how you can present your example without dirty-equals:

```python
def test_function():
    result = get_data()
    assert result == snapshot(
        {"date": result["date"], "payload": "some data"}
    )
    assert isinstance(result["date"], datetime.datetime)
```

Reply:
> dirty-equals are handled special when `--inline-snapshot=update` is used (they are preserved). This is not the case for your examples.

What's the motivation for this special case?

> I intentional decided to limit the documentation/implementation to dirty-equals, because I did not know how a general use case might look like or what the problem could be that the user might try to solve.

What about the examples above?

> I would prefer to implement features when I understand the usage scenario the user might have. But I have in general nothing against it, just in a later version maybe.

What's the usage scenario for `update` and how does it relate to dirty-equals vs other expressions? I haven't had a need for `update` myself.

You make it sound like this has only been implemented for dirty-equals, but doesn't `fix` work equally well for all kinds of expressions?

I think it's important for users to know about the general feature. It's useful information and not everyone will want to use dirty-equals. As I've said, some of my own use cases involve non-dirty-equals expressions.

Reply:
The idea behind update is that the `__repr__()` implementation of some object might change (defaults are excluded from the repr, the class was renamed in a backward compatible way and is available with the new and old name, or you updated to a newer inline-snapshot version which creates multiline strings).

update changes the cases where old_value==new_value but repr(new_value) is different from the current source.

I have currently no way to decide if the user changed the code (like in your example) or if __repr__() was changed. 
This is the reason why I need a way to detect if update should change the code (the current implementation checks it with `isinstance(old_value,DirtyEquals)` ).

> What about the examples above?

They did not looked like to me like examples which you have created to have some examples. Sorry if this is not the case.

> As I've said, some of my own use cases involve non-dirty-equals expressions.

Thank you for mentioning this again. I think you mean some `[...]*10` cases right?
My best idea currently is to wrap these changes in something like `no_update([...]*10)` or to accept the fact that an update would revert your changes.

Reply:
hi @alexmojaki, I created an example which shows how how this could be implemented with dirty-equals.

``` python
from inline_snapshot import snapshot
from dirty_equals import DirtyEquals


class NoSnapshotUpdate(DirtyEquals):
    def __init__(self, value):
        self.__value = value
        super().__init__(value=value)

    def equals(self, other):
        return other == self.__value


def test_blub():
    assert {
        1: 5,
        2: 2,
    } == snapshot(
        {
            1: 1,
            2: NoSnapshotUpdate(1 + 1),
        }
    )
```
The 1+1 will not be updated, even if you run it with update

``` bash
pytest --inline-snapshot=fix,update
```

you can either use something like this, or never use `update`. I see no other way to combine both worlds.

I would also like to stick with the current dirty-equals support. It could become to complicated to support generic solution which might be used in a variety of ways.

But also thank you that you convinced me to support dirty-equals. I'm really happy with this idea :+1: .

Reply:
I don't use `update` and I don't think its exact behaviour is that important, I was just surprised that dirty-equals was being treated specially.

I still think it's very useful for readers to know that `fix` preserves other expressions, but that's just my opinion and definitely not something that needs to block this PR.